src/lighthouse.hpp

```diff
@@ -19,7 +19,7 @@ class Lighthouse {
 	void rotate_towards_cursor(glm::vec2 cursor_pos);
 	void render(le::Renderer& renderer) const;
 	void check_visibility_range(Enemy& enemy);
-	void check_damage_taken(Enemy& enemy);
+	void check_damage_taken(float enemy_diameter, float dist_sq);
```

Comment:
The idea with the Enemy reference was to accommodate future extensible behavior, depending on the enemy and light type

Reply:
It can be updated when that happens. The way it was before this PR was wasting time on redundant computation. In general that's not a _huge_ deal but this is hot code: runs every frame, so you want to be as efficient as you can (without going overboard / trying to be too clever).